src/main/java/address/browser/AddressBookBrowser.java

```diff
@@ -0,0 +1,273 @@
+        return tab;
+    }
+
+    /**
+     * Registers listeners for automating clicking and scrolling.
+     */
+    public void registerListeners() {
+        for (int i = 0; i < pages.size(); i++) {
+            pages.get(i).addLoadListener(new LoadAdapter() {
+                @Override
+                public void onFinishLoadingFrame(FinishLoadingEvent finishLoadingEvent) {
+                    if (finishLoadingEvent.isMainFrame()) {
+                        automateClickingAndScrolling(finishLoadingEvent.getBrowser());
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Returns the UI view of the AddressBookBrowser.
+     * @return
+     */
+    public TabPane getAddressBookBrowserView(){
+        return addressBookBrowserView;
+    }
+
+    /**
+     * Dispose the resources used by this browser. Once called, the browser is no longer usable.
+     */
+    public void dispose(){
+        for (int i = 0; i < pages.size(); i++) {
+            pages.get(i).dispose();
+        }
+    }
+
+    /**
+     * Loads the person's profile page to the browser.
+     * It also performs caching of other person profile page.
+     * PreCondition: getFilteredPersons().size() >= 1
+     * @param person
+     */
+    public void loadProfilePage(Person person) {
+
+        List<Person> listOfContactsDisplayed = getFilteredPersons();
+        int indexOfPersonInListOfContacts = listOfContactsDisplayed.indexOf(person);
+        Range rangeOfIndexToLoad = getRangeOfIndexToLoad(listOfContactsDisplayed, indexOfPersonInListOfContacts);
+
+        ArrayList<Person> listOfPersonsToBeLoaded = getListOfPersonsToBeLoaded(listOfContactsDisplayed,
+                                                                               indexOfPersonInListOfContacts);
+        unloadProfilePages(listOfContactsDisplayed, rangeOfIndexToLoad);
+        ArrayList<Person> listOfAlreadyLoadedPerson = getListOfAlreadyLoadedPerson(listOfContactsDisplayed,
+                                                                                   rangeOfIndexToLoad);
+        listOfPersonsToBeLoaded.removeAll(listOfAlreadyLoadedPerson);
+        loadProfilePages(listOfPersonsToBeLoaded);
+        selectTabAssignedForPerson(person);
+    }
+
+    /**
+     * Computes the range of index of the displayed AddressBook contact that needs to be added to the browser cache.
+     * @param listOfContactsDisplayed The list of contacts currently displayed.
+     * @param indexOfPersonInListOfContacts The index of the currently selected person in the AddressBook contact list.
+     * @return
+     */
+    private Range getRangeOfIndexToLoad(List<Person> listOfContactsDisplayed, int indexOfPersonInListOfContacts) {
+        Range rangeOfIndexToLoad;
+        if(BrowserManager.NUMBER_OF_PRELOADED_PAGE >= listOfContactsDisplayed.size()){
+            rangeOfIndexToLoad = new Range(indexOfPersonInListOfContacts, indexOfPersonInListOfContacts -1);
```

Comment:
new Range(indexOfPersonInListOfContacts, indexOfPersonInListOfContacts -1); 
and 
new Range(0, listOfContactsDisplayed.size() - 1)

are two different values?